### DISCUSSION TITLE
How to flash Bolt 1.1 all firmware ?

### DISCUSSION THREAD

**Discussion by justinleeyang** [bitcraze/discussions#1720] [General]
Hi,

How to flash Bolt 1.1 all firmware compared with cf21， include nRF51822 and STM32F405?

**Comment by ToveRumar** [comment]
Hi @justinleeyang !

If you are using the cfclient you can flash by clicking "Connect"->"Bootloader" 

![image](https://github.com/user-attachments/assets/bb819002-d2ec-4aa7-ac26-40185b1a0648)

Choose the platform "bolt", and press program. This will flash the latest FW for all onboard chips, and all connected decks. 

I hope this answered your question!

**Reply by justinleeyang** [reply]
if my device is crazyflie2.1, and I want to flash the Bolt 1.1 firmware. Is this possible by upgrading according to your method?

**Reply by ToveRumar** [reply]
No, the stock firmware will not work on another platform. You could manually modify the binary to run on a crazyflie, however it is not recommended. Could you please describe what you would like to achieve so that I could try to recommend a good way of doing this. 

In this case you need to build your own SW and flash this using cfloader (https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/building-and-flashing/build/).
Please refer to the documentation on how to build and modify the firmware (https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/).
